generators/add-licence-headers/index.js

```diff
@@ -0,0 +1,202 @@
+    });
+  }
+
+  prompting() {
+    return this.prompt([
+      {
+        type: 'input',
+        name: 'licencePath',
+        message: `Please enter the relative path to the ${chalk.blue(defaultLicenceFileName)} file`,
+        when: this.options.licencePath === defaultLicencePath
+      },
+      {
+        type: 'checkbox',
+        name: 'excludedFileTypes',
+        message: 'Exclude file types from adding headers',
+        choices: Object.keys(comments),
+        when: this.options.excludedFileTypes.length === 0
+      }
+    ]).then((props) => {
+      this.licencePath = props.licencePath || this.options.licencePath;
+      const excludedFileTypes = props.excludedFileTypes || this.options.excludedFileTypes.split(',');
+
+      // filter out excluded file types
+      this.fileTypes = Object.keys(comments).filter((type) => !excludedFileTypes.includes(type));
+
+      // TODO: How to abort the generator correctly?
+      return this._readLicenceFile();
+    }).catch((err) => this.log(err));
+  }
+
+  writing() {
+    this._generateComments();
+
+    try {
+      const sourceFolders = this._getSourceFolders();
+
+      sourceFolders.forEach((folderName) => {
+        const pluginPath = this.destinationPath(folderName);
+        if (fs.existsSync(pluginPath)) {
+          this.fileTypes.forEach((type) => this._addComments(pluginPath, type));
+        }
+      });
+    } catch (e) {
+      this._abort(e);
+    }
+  }
+
+  _readLicenceFile() {
+    try {
+      this.licenceText = this.fs.read(this.licencePath);
+    } catch (e) {
+      return this._abort(e);
+    }
+  }
+
+  _generateComments() {
+    // get maximum line length by looping through all lines, returning the line length and passing them to Math.max
+    const lineArray = this.licenceText.split('\n');
+    const maxLineLength = Math.max(...lineArray.map((line) => line.length));
+
+    const align = (string, threshold, timesSpaces) => {
+      if (!timesSpaces) {
+        timesSpaces = threshold;
+      }
+
+      if (string.length > threshold) {
+        return ' '.repeat(timesSpaces);
+      }
+      return '';
+    };
+
+    Object.keys(comments).forEach((fileType) => {
+      const commentStyle = comments[fileType];
+      let comment = commentStyle.begin + commentStyle.body.repeat(maxLineLength) + '\n';
+      lineArray.forEach((line) => {
+        // only add a space character when the opening comment contains more than one charcters (e.g. to align the asterisks in .ts files vs. aligning the hashes in .py files)
+        comment += align(commentStyle.begin, commentStyle.aligningSpaces);
+        comment += `${commentStyle.body}${align(line, 0, 1) + line}\n`;
+      });
+
+      comment += align(commentStyle.begin, commentStyle.aligningSpaces);
+      comment += `${commentStyle.body.repeat(maxLineLength)}${commentStyle.end}`;
+      this.comments[fileType] = comment;
+    });
+  }
+
+  _getSourceFolders() {
+    const path = this.destinationPath().split('/');
```

Comment:
you could use some utility function from the `fs` or `path` module instead. Don't know regarding windows vs unix